### DISCUSSION TITLE
searchble

### DISCUSSION THREAD

**Discussion by Shin-Yu-1** [Swift-Coding-Club/Team_Nav#23] [STUDY WITH ME]
금일 searchble에 대해서 좀 알아보았고 이에 대해 간략히 글을 써봅니다.

#  👉 searchable
<img width="280" alt="스크린샷 2023-04-27 오전 11 37 37" src="https://user-images.githubusercontent.com/82097725/234745590-863fe44e-b17c-4818-83b6-83fa20394b25.png">

해당 뷰에서 검색 가능하도록 검색 필드를 표시합니다.
이전에는 TextField를 활용한 것으로 생각됩니다만, SwiftUI에서는 searchable modifier을 통해 구현가능합니다.

# 👉 적용 방법
검색 기능이 필요한 뷰에 적용하게 됩니다.
내비게이션 바 제목 아래에 검색바가 위치합니다.
이 기능을 위해서는 NavigationView가 필요합니다.

### 코드 예시
```
    ... 생략 ...

    @State var searchQueryString = ""

    var body: some View {
        NavigationView {

            ... 생략 ...

            )
            .navigationTitle("Titile")
            .searchable(
                text: $searchQueryString,
                placement: .navigationBarDrawer,
                prompt: "검색"
            )
        }
    }
```
- **text**
검색 필드에 입력하는 문자열
searchQueryString에 바인딩함.

- **placement**
검색 필드 위치 지정. 
기본값은 .automatic
.navigationBarDrawer를 사용하여 검색 필드를 네비게이션 바(drawer)의 아래에 배치하도록 지정함. 

- **prompt**
검색 필드에 표시될 힌트.
html 엘리먼트 중 input의 placeholder와 같은 기능을 함.

추가로 더 필요한 내용이 있다면 알려주세요!

## 참고 링크
- https://developer.apple.com/documentation/swiftui/divider/searchable(text:placement:prompt:)-8i0vn/
- https://www.kodeco.com/26924502-swiftui-search-getting-started

**Comment by LiSeul** [comment]
대단해여!!! 역시!!!!!!!!!!!!!!! 👍👍👍👍👍👍
